src/v/kafka/server/request_context.h

```diff
@@ -49,6 +50,14 @@ struct request_header {
 std::ostream& operator<<(std::ostream&, const request_header&);
 
 class request_context {
+private:
+    template<typename T, typename = int>
+    struct has_throttle_member : std::false_type {};
+
+    template<typename T>
+    struct has_throttle_member<T, decltype((void)T::throttle_time_ms, 0)>
+      : std::true_type {};
```

Comment:
we can use concepts now. so this is something like

```
template<typename T>
concept has_throttle_time_ms = requires(T t) {
   // member check
   // chrono type check
}
```

then I think you can do `if constexpr (has_throttle_time_ms ...)`